x-pack/plugins/infra/public/pages/logs/log_entry_rate/use_log_entry_rate_results_url_state.tsx

```diff
@@ -40,36 +49,102 @@ export const useLogAnalysisResultsUrlState = () => {
   const [getTime] = useKibanaTimefilterTime(TIME_DEFAULTS);
   const { from: start, to: end } = getTime();
 
-  const [timeRange, setTimeRange] = useUrlState({
-    defaultState: {
+  const defaultTimeRangeState = useMemo(() => {
+    return {
       startTime: start,
       endTime: end,
-    },
-    decodeUrlState: (value: unknown) =>
-      pipe(urlTimeRangeRT.decode(value), fold(constant(undefined), identity)),
+    };
+  }, [start, end]);
+
+  const [timeRange, _setTimeRange] = useUrlState({
```

Comment:
Nit: How about this so we don't have the switch of meaning between this and the return value names?

```suggestion
  const [urlTimeRange, setUrlTimeRange] = useUrlState({
```